### PR TITLE
testing javascript

### PR DESCRIPTION
**Pull request by yochiwarez**
This PR contains the following functions :

- capitalize(string) takes a string and returns that string with the first character capitalized.

- reverseString(string) takes a string and returns it reversed.

- A calculator object that contains the basic operations: add, subtract, divide, and multiply.

- Caesar Cipher. Read about it on this website

- Array Analysis. Write a function that takes an array of numbers and returns an object with the following properties: average, min, max, and length.

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    for (let i = s.length - 1; i > -1; i -= 1) {
+      answer += s.charAt(i);
+    }
+    return answer;
+  },
+  Calculator: class {
+    constructor(n1, n2) {
+      this.n1 = n1;
+      this.n2 = n2;
+    }
```
Comment (Oghenebrume50) [inline functions.js]: We want to be able to call add, sub, .... with different number values  so initialize the constructor should be empty while the methods in the class should accept parameters

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    multiply() { return this.n1 * this.n2; }
+  },
+  caesar: (s, ofs) => Array.from(s).reduce((acc, ch) => {
+    const i = (parseInt(ch, 36) + ofs) % 36;
+    let ne = (!i * 10 + i).toString(36);
+    ne = ne === 'NaN' ? ' ' : ne;
+    return acc + ne;
+  }, ''),
+  analyze: array => {
+    const averageVal = array.reduce((acc, i) => acc + i);
```
Comment (Oghenebrume50) [inline functions.js]: This just sums up the value it does not calculate the average

File: functions.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    divide() { return this.n1 / this.n2; }
+
+    multiply() { return this.n1 * this.n2; }
+  },
+  caesar: (s, ofs) => Array.from(s).reduce((acc, ch) => {
+    const i = (parseInt(ch, 36) + ofs) % 36;
+    let ne = (!i * 10 + i).toString(36);
+    ne = ne === 'NaN' ? ' ' : ne;
+    return acc + ne;
+  }, ''),
```
Comment (Oghenebrume50) [inline functions.js]: This does not work correctly with punctuations and cases  see this 
we want to preserve the case and also keep the punctuations

File: functions.js (hunk 1 above)
Comment (yochiwarez) [reply functions.js]: Calculator: class {
    add(n1, n2) { return n1 + n2; }

    substract(n1, n2) { return n1 - n2; }

    divide(n1, n2) { return n1 / n2; }

    multiply(n1, n2) { return n1 * n2; }
  }

File: functions.js (hunk 2 above)
Comment (yochiwarez) [reply functions.js]: const averageVal = (array.reduce((acc, i) => acc + i)) / array.length;

File: functions.js (hunk 3 above)
Comment (yochiwarez) [reply functions.js]:
```
caesar: (s, ofs) => Array.from(s).reduce((acc, ch) => {
    let ne;
    if (ch.match(/[.,:;!?¿#%=&@\\+()[\]"'´`/*¡_-]/)) {
      ne = ch;
    } else {
      const i = (parseInt(ch, 36) + ofs) % 36;
      ne = (!i * 10 + i).toString(36);
      ne = ne === 'NaN' ? ' ' : ne;
    }
    return acc + ne;
  }, ''),

**Pull request merged**